### ISSUE TITLE
Localization issue with PortableObjectStringLocalizerFactory

### ISSUE THREAD

**Issue by hishamco** [OrchardCMS/OrchardCore#5732]
I'm still worried how the `PortableObjectStringLocalizerFactory` can inject `IOptions<RequestLocalizationOptions>` if the `RequestLocalizationOptions` are set in the `Configure` method. I'm sure code like this https://github.com/aspnet/Localization/blob/master/samples/LocalizationSample/Startup.cs#L27-L37 will not work properly, because the `PortableObjectStringLocalizerFactory` doesn't know about the `RequestLocalizationOptions` changes that's made in `Configure` method. Am I right? @jtkech

**Comment by jtkech** [comment]
Regarding your link, they do at the app level what we do at the tenant level through the `OC.Localization` module startup configure method. This is here that we update `RequestLocalizationOptions` and that we call `UseRequestLocalization()`.

But to be executed the module needs to be enabled, then, based on the current `LocalizationSettings` that we can udate through the admin, we override some options properties as default and supported cultures. Note: So another module can only configure other properties as `FallBackToParentCultures` e.g by using `services.Configure<RequestLocalizationOptions>`.

Hmm, so to use it during a setup, maybe we would need to add `OC.Localization` to the minimum set of features that are enabled for a setup. And then, because there is not yet `LocalizationSettings`, the `LocalizationService` would have to check this and then try to fallback to the cultures that may have been defined through the configuration (e.g appsettings.json).

**Comment by hishamco** [comment]
> This is here that we update RequestLocalizationOptions and that we call UseRequestLocalization()

Yep, but it called on `ConfigureServices()` using `services.Configure<RequestLocalizationOptions>` which is fine in this case

I'm thinking if I can a new AspNetCore app I reference the OC to use the `PortableObjectStringLocalizer` with settings like the link I mentioned before, in this case the `PortableObjectStringLocalizerFactory` will unable to read my `RequestLocalizationOptions`

I faced the same issue in my `JsonStringLocalizer` [here](https://github.com/hishamco/My.Extensions.Localization.Json/pull/41), that's why I register it similar to what we did in the localization module
